### PR TITLE
Feat/age5 modifications

### LINKED ISSUE #105: Add age5 data
## Feature Request
**Is your feature request related to a problem? Please describe.**
Following the [forum post on the `age5`](https://forum.morpho.xyz/t/mip-morpho-rewards-age-5/303), the script has to implement aave v3 ETH market

**Describe the solution you'd like**

- add morpho aave v3 indexing in the subgraph. Only ETH market has to be implemented, so no need to index collateral events for now
- add MarketMinimal fetching to compute the age distribution for eth market.
- add emission script for age5
**Describe alternatives you've considered**

## Are you willing to resolve this issue by submitting a Pull Request?

- [x] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

### PR DESCRIPTION
**Pull request by julien-devatom**
### Description of change
- Add ma3 ETH marked
- Add `age5` emission script following the vote proposal

Closes: #105 

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] `yarn lint` passes with this change
- [x] `yarn test` passes with this change
- [x] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [x] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,13 @@ Morpho Rewards can only be run on Node 18 and higher, notably due to its depende
 
 ## **Get Started**
 
-- Install node dependencies using `yarn`:
+- make sure you are using the correct node version (>= 18)
+```bash
+nvm use
```
Comment (MerlinEgalite) [inline README.md:13]:
```suggestion
nvm use 18
```

no?

File: src/ages/distributions/ageFiveDistribution/marketsRepartition.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    weight: BigNumber.from(1014),
+  },
+  // Morpho Compound markets
+  cUSDT: {
+    market: "0xf650C3d88D12dB855b8bf7D11Be6C55A4e07dCC9",
+    weight: BigNumber.from(274),
+  },
+  // Morpho AaveV3 markets
+  aWETHV3: {
+    market: "0x030bA81f1c18d280636F32af80b9AAd02Cf0854e",
```
Comment (MerlinEgalite) [inline src/ages/distributions/ageFiveDistribution/marketsRepartition.ts:36]: ma3 workds with underlying addresses, is this still ok?

**Review by MerlinEgalite — COMMENTED** [review]
LGTM but one question

File: README.md (hunk 1 above)
Comment (julien-devatom) [reply README.md:13]: it fallback on the `.nvmrc` by default

File: src/utils/markets/fetchMarketsData.ts [hunk 3]
```diff
@@ -19,20 +23,89 @@ import { MarketMinimal } from "../graph/getGraphMarkets/markets.types";
...
+    provider
+  );
+
+  const oracle = AaveV3Oracle__factory.connect(await addressesProvider.getPriceOracle(overrides), provider);
+
+  const allMarkets = await morpho.marketsCreated(overrides);
+
+  return Promise.all(
+    allMarkets.map(async (underlying) => {
+      const aToken = AToken__factory.connect(underlying, provider);
```
Comment (oumar-fall) [inline src/utils/markets/fetchMarketsData.ts:62]: ⚠️ underlying !== aToken

File: src/utils/markets/fetchMarketsData.ts [hunk 4]
```diff
@@ -19,20 +23,89 @@ import { MarketMinimal } from "../graph/getGraphMarkets/markets.types";
...
+      const [decimals, totalPoolSupply, poolSupplyAmount] = await Promise.all([
+        aToken.decimals(overrides),
+        aToken.totalSupply(overrides),
+        aToken.balanceOf(morpho.address, overrides),
+      ]);
+      const { variableDebtTokenAddress } = await lendingPool.getReserveData(underlying, overrides);
```
Comment (oumar-fall) [inline src/utils/markets/fetchMarketsData.ts:68]: you should get the aToken address here as well

File: src/utils/markets/fetchMarketsData.ts [hunk 5]
```diff
@@ -62,14 +135,15 @@ const getAaveMarketsParameters = async (snapshotBlock: providers.BlockTag, provi
         { p2pBorrowAmount, poolBorrowAmount, p2pSupplyAmount, poolSupplyAmount },
       ] = await Promise.all([
         debtToken.totalSupply(overrides),
-        oracle.getAssetPrice(underlying, overrides),
+        oracle.getAssetPrice(underlying, overrides).then((price) => price.mul(10 ** 10)), // from 8 to 18 decimals
```
Comment (oumar-fall) [inline src/utils/markets/fetchMarketsData.ts:138]: on aaveV2 price are given with 18 decimals no?

File: src/utils/markets/fetchMarketsData.ts [hunk 6]
```diff
@@ -19,20 +23,89 @@ import { MarketMinimal } from "../graph/getGraphMarkets/markets.types";
...
+          },
+        },
+        {
+          supply: { p2pIndex: p2pSupplyIndex },
+          borrow: { p2pIndex: p2pBorrowIndex },
+        },
+        poolBorrowAmount,
+      ] = await Promise.all([
+        debtToken.totalSupply(overrides),
+        oracle.getAssetPrice(underlying, overrides),
```
Comment (oumar-fall) [inline src/utils/markets/fetchMarketsData.ts:87]: and here it's with 8 decimals

**Review by oumar-fall — CHANGES_REQUESTED** [review]

File: src/utils/markets/fetchMarketsData.ts (hunk 3 above)
Comment (julien-devatom) [reply src/utils/markets/fetchMarketsData.ts]: fixed

**Review by oumar-fall — APPROVED** [review]

**Review by ghivert — APPROVED** [review]

**Pull request merged**